### DISCUSSION TITLE
tfrecord

### DISCUSSION THREAD

**Discussion by ansaricard** [serizba/cppflow#165] [Q&A]
hello every body
according to the examples I can load a model or read an image and predict raw images with model in C++ codes
what shall I do to read tfrecord files and parse examples or parse tensors have been extracted from tfrecord files?
how can I read tfrecord files? 
tensorflow  C++ api documents are weak but cppflow solution was fantastic, thanks

**Comment by serizba** [comment]
Hi!,

Sorry for the late reply. I've never tried, but I guess it should be doable with [TFRecordDataset](https://www.tensorflow.org/api_docs/python/tf/raw_ops/TFRecordDataset) and [ParseSingleExample](https://www.tensorflow.org/api_docs/python/tf/raw_ops/ParseSingleExample), that have their equivalents in cppflow raw_ops.
